ui/src/components/forms/SegmentsPicker.tsx

```diff
@@ -1,34 +1,42 @@
-import { MinusSmallIcon } from '@heroicons/react/24/outline';
-import { useRef } from 'react';
+import { MinusSmallIcon, PlusSmallIcon } from '@heroicons/react/24/outline';
+import { useRef, useState } from 'react';
 import Combobox from '~/components/forms/Combobox';
 import { FilterableSegment, ISegment } from '~/types/Segment';
 import { truncateKey } from '~/utils/helpers';
 
 type SegmentPickerProps = {
+  editMode?: boolean;
   segments: ISegment[];
   selectedSegments: FilterableSegment[];
   segmentAdd: (segment: FilterableSegment) => void;
   segmentReplace: (index: number, segment: FilterableSegment) => void;
   segmentRemove: (index: number) => void;
 };
 
-export default function SegmentsPicker(props: SegmentPickerProps) {
-  const {
-    segmentAdd,
-    selectedSegments: parentSegments,
-    segments,
-    segmentRemove,
-    segmentReplace
-  } = props;
+export default function SegmentsPicker({
+  editMode = false,
```

Comment:
One thing I realized is we need to handle `readOnly` mode (when using git/fs/s3 backends) and disable this component from allowing any segments to be added.

we do this elsewhere in the app, like here's the `FlagForm` component:

https://github.com/flipt-io/flipt/blob/main/ui/src/components/flags/FlagForm.tsx

```
  const readOnly = useSelector(selectReadonly);
```

```
  <Toggle
    id="enabled"
    name="enabled"
    label="Enabled"
    disabled={readOnly} // pass in as `disabled` to the Toggle component
    checked={enabled}
    onChange={(e) => {
      formik.setFieldValue('enabled', e);
    }}
  />
```

Then the Toggle component handles things like showing the not allowed cursor when hovering over action buttons:

https://github.com/flipt-io/flipt/blob/main/ui/src/components/forms/Toggle.tsx#L15C1-L48C8


we should also handle this in the Rollout and Rule Forms so that we don't allow any data changes when in readonly mode.

Reply:
@markphelps Did that in this commit: https://github.com/flipt-io/flipt/pull/1953/commits/87d09cdf37d272453292fcf9e25529437cfba8d8

Thank you!